backends-velox/src/main/scala/org/apache/spark/sql/execution/datasources/velox/VeloxFormatWriterInjects.scala

```diff
@@ -80,7 +89,7 @@ trait VeloxFormatWriterInjects extends GlutenFormatWriterInjectsBase {
 
       override def close(): Unit = {
         writeQueue.close()
-        datasourceJniWrapper.close(instanceId)
+        datasourceJniWrapper.close(executionCtxHandle, dsHandle)
```

Comment:
Such change makes code slightly more complicated than before (2 params vs 1 params). We'd find a better way later on.

Reply:
Another solution is make each JniWrapper methods be member methods instead of static, so we can hide executionCtxHandle into instance, but it need lots changes for exist all JniWrapper.

Reply:
Yes so let's keep the code as it is for now.

Reply:
can you use c++ `thread_local ` like what java did ? I think the `ExecutionCtx` should be got directly at c++ side without any handle.

Reply:
I dont really want to bind ExecutionCtx with thread, and we found PythonRunner will share same Spark TaskContext in different thread before.

Reply:
> we found PythonRunner will share same Spark TaskContext in different thread before.

It's not about this pr, just curious how can it happen.

Besides, I do not think they are the same thing. Unlike the python using a new process, the jni side is just a part of jvm evaluator, and they have to be in the same thread.

Reply:
Indeed it happened, you can try debug with a python UDF.